### ISSUE TITLE
Project name inconsistent between codewind and eclipse

### ISSUE THREAD

**Issue by jopit** [eclipse/codewind#862]
**Codewind version:** latest
**OS:** macOS Catalina

I entered the following in the create project wizard. It created a project which was imported into eclipse with the name 'project', but in the codewind view, the project is called 'liberty2'

<img width="373" alt="Screen Shot 2019-10-25 at 10 31 10 AM" src="https://user-images.githubusercontent.com/29612372/67580662-b2a3da00-f714-11e9-8797-15d2e384339c.png">

If you do a similar thing in the eclipse create java project wizard, the project in eclipse would be given the name provided, not given the name of the directory.

**Comment by eharris369** [comment]
Unfortunately the SmartImportJob does not provide a way to give the project a name (even the smart import wizard does not have this).  I tried calling setListener(RecursiveImportListener) on SmartImportJob but that did not work (not really what it was meant for anyway).  A workspace resource change listener could be added and the project refactored but the project would show initially with the original name.  It might show that way for a while for a Maven project if a lot of Maven jobs get kicked off on project import and the refactoring job does not get run right away.

There is also the situation where the user changes the Eclipse project name - should the Codewind project name change as well?  There is no API for this but it could be handled by the IDE.

I also found that in this case if the project is removed from Codewind, and the user requests to delete it as well, the Eclipse project does not get deleted - this is because it is looking the project up by name instead of location.
